### PR TITLE
PHP 8.0 Support: Attribute Syntax (Part 4)

### PR DESCRIPTION
**Pull request by junichi11**
- https://github.com/apache/netbeans/issues/4300
- https://wiki.php.net/rfc/attributes_v2
- https://wiki.php.net/rfc/shorter_attribute_syntax
- https://wiki.php.net/rfc/shorter_attribute_syntax_change
- Fix the code completion feature
- Index classess marked as the Attribute(`#[\Attribute]`)
- Add/Fix unit tests for the indexer and the code completion
- Increment the spec version

nb-php-gh-4300-attribute-cc

Note: The following case is not fixed yet.
```php
 #[MyAttr(self::^CONST1)] // don't work here yet
 class Example {
     public const CONST1 = 1;
 }
```

### CODE REVIEW

File: php/php.editor/src/org/netbeans/modules/php/editor/elements/IndexQueryImpl.java [hunk 1]
```diff
@@ -1732,6 +1743,18 @@ public boolean isAccepted(PhpElement element) {
         };
     }
 
+    private static ElementFilter forAttributeClass() {
+        return new ElementFilter() {
+            @Override
+            public boolean isAccepted(PhpElement element) {
+                if (element instanceof ClassElementImpl) {
+                    ClassElementImpl classElement = (ClassElementImpl) element;
```
Comment (tmysik) [inline php/php.editor/src/org/netbeans/modules/php/editor/elements/IndexQueryImpl.java:1751]: Is this code complete? I mean, I would expect some operation on `classElement`, no?

**Review by tmysik — APPROVED** [review]
The change is not small but looks OK to me. Also, as always, we have plenty of tests, thanks!

File: php/php.editor/src/org/netbeans/modules/php/editor/elements/IndexQueryImpl.java (hunk 1 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/elements/IndexQueryImpl.java:1751]: Oops, this method is not used. Maybe..., I tried filtering attribute classes from all classes. But I stopped doing that and forgot to remove this...
I'll remove this. Thank you for catching it!

**Comment by junichi11** [comment]
@tmysik Thank you for your review and your time! I'll merge this after I remove the unused methods :)

File: php/php.editor/src/org/netbeans/modules/php/editor/elements/IndexQueryImpl.java (hunk 1 above)
Comment (junichi11) [reply php/php.editor/src/org/netbeans/modules/php/editor/elements/IndexQueryImpl.java]: Fixed: https://github.com/apache/netbeans/compare/f88db839e63af80394ac927013ddfb3d2f2adfca..4b4b377b873b753e1bac6fa90d926b366633a01c

**Comment by tmysik** [comment]
@junichi11 No need to thank me, you did all the work :)

**Pull request merged**